src/main/java/com/conveyal/analysis/controllers/RegionalAnalysisController.java

```diff
@@ -406,17 +409,26 @@ private UrlWithHumanName getRegionalResults (Request req, Response res) throws I
         // For newer analyses that have multiple cutoffs, percentiles, or destination pointsets, these initial values
         // are coming from deprecated fields, are not meaningful and will be overwritten below from query parameters.
         int percentile = analysis.travelTimePercentile;
-        int cutoffMinutes = analysis.cutoffMinutes;
+        int threshold = analysis.cutoffMinutes;
         String destinationPointSetId = analysis.grid;
 
-        // Handle newer regional analyses with multiple cutoffs in an array.
-        // If a query parameter is supplied, range check it, otherwise use the middle value in the list.
-        // The cutoff variable holds the actual cutoff in minutes, not the position in the array of cutoffs.
-        if (analysis.cutoffsMinutes != null) {
+        if (analysis.request.includeTemporalDensity) {
+            int nThresholds = analysis.request.dualAccessThresholds.length;
+            int[] thresholds = analysis.request.dualAccessThresholds;
+            checkState(nThresholds > 0, "Regional analysis has no thresholds.");
+            threshold = getIntQueryParameter(req, "threshold", thresholds[nThresholds / 2]);
+            checkArgument(new TIntArrayList(thresholds).contains(threshold),
+                    "Dual access thresholds for this regional analysis must be taken from this list: (%s)",
+                    Ints.join(", ", thresholds)
+            );
+        } else if (analysis.cutoffsMinutes != null) {
+            // Handle newer regional analyses with multiple cutoffs in an array.
+            // If a query parameter is supplied, range check it, otherwise use the middle value in the list.
+            // The cutoff variable holds the actual cutoff in minutes, not the position in the array of cutoffs.
```

Comment:
Should this comment (specifically the part about using the middle value in the list) be moved up (as `[nThresholds / 2]` is also used above)?